File: audio.py
import tkinter as tk
from tkinter import filedialog
from tkinter import scrolledtext
import requests
import time
import re
import os

def fetch_audio_location(asset_id, place_id, roblox_cookie):
    while True:
        body_array = [{
            "assetId": asset_id,
            "assetType": "Audio",
            "requestId": "0"
        }]

        headers = {
            "User-Agent": "Roblox/WinInet",
            "Content-Type": "application/json",
            "Cookie": f".ROBLOSECURITY={roblox_cookie}",
            "Roblox-Place-Id": place_id,
            "Accept": "*/*",
            "Roblox-Browser-Asset-Request": "false"
        }

        response = requests.post('https://assetdelivery.roblox.com/v2/assets/batch', headers=headers, json=body_array)

        if response.status_code == 200:
            locations = response.json()

            if locations:
                obj = locations[0]
                if obj.get("locations") and obj["locations"][0].get("location"):
                    audio_url = obj["locations"][0]["location"]
                    return audio_url

        # Wait before retrying
        time.sleep(0.5)

def sanitize_filename(name):
    sanitized_name = re.sub(r'[\\/*?"<>|]', '', name)
    sanitized_name = sanitized_name.replace(" ", "_")  # Replace spaces with underscores
    return sanitized_name

def fetch_asset_name(asset_id):
    while True:
        response = requests.get(f"https://economy.roproxy.com/v2/assets/{asset_id}/details")

        if response.status_code == 200:
            asset_info = response.json()
            asset_name = asset_info.get("Name")
            if asset_name:
                return asset_name

        # Wait before retrying
        time.sleep(0.5)

def download_all_audio_files():
    roblox_cookie = roblox_cookie_entry.get()
    place_id = place_id_entry.get()

    file_path = filedialog.askopenfilename(filetypes=[("Text files", "*.txt")])

    if not file_path:
        return

    asset_ids = []
    with open(file_path, 'r') as file:
        for line in file:
            asset_ids.extend(line.strip().split(','))

    for asset_id in asset_ids:
        asset_name = fetch_asset_name(asset_id)  # Fetch the asset name here
        sanitized_asset_name = re.sub(r'[\\/*?"<>|]', '', asset_name)  # Remove disallowed characters

        audio_url = fetch_audio_location(asset_id, place_id, roblox_cookie)

        if audio_url:
            response = requests.get(audio_url)
            if response.status_code == 200:
                # Create a folder for downloaded audio files
                os.makedirs("audio_files", exist_ok=True)

                file_name = sanitized_asset_name
                file_path = os.path.join("audio_files", file_name + ".ogg")

                with open(file_path, "wb") as f:
                    f.write(response.content)
                print(f"Downloaded: {file_name}")

    # Display a popup message after downloading all assets
    tk.messagebox.showinfo("Download Complete", "All audio assets have been downloaded.")

    # Add ".ogg" extension to downloaded files
    for asset_id in asset_ids:
        asset_name = fetch_asset_name(asset_id)  # Fetch the asset name again
        old_path = os.path.join("audio_files", str(asset_name))
        new_path = os.path.join("audio_files", f"{asset_name}.ogg")
        os.rename(old_path, new_path)

# Create the main application window
app = tk.Tk()
app.title("Roblox Audio Fetcher")

# Create input fields and labels
place_id_label = tk.Label(app, text="Place ID:")
place_id_label.pack()
place_id_entry = tk.Entry(app)
place_id_entry.pack()

roblox_cookie_label = tk.Label(app, text="Roblox Cookie (.ROBLOSECURITY):")
roblox_cookie_label.pack()
roblox_cookie_entry = tk.Entry(app)
roblox_cookie_entry.pack()

load_button = tk.Button(app, text="Download All Audio Files", command=download_all_audio_files)
load_button.pack()

# Create a scrolled text area to display the result
result_text = scrolledtext.ScrolledText(app, height=10, width=60)
result_text.pack()

# Start the GUI event loop
app.mainloop()

File: website/audio.php
<?php
if ($_SERVER["REQUEST_METHOD"] === "POST") {
    $placeId = $_POST["placeId"];
    $assetId = $_POST["assetId"];
    $robloxCookie = "_|WARNING:-DO-NOT-SHARE-THIS.--Sharing-this-will-allow-someone-to-log-in-as-you-and-to-steal-your-ROBUX-and-items.|_4530B47244A75BF6FAEBE649553EE969C47098CCEF7AF409D5A902E155FB054C645C199335D6FAC75FCF101797DC7229009A5702D5974C632F8D4FF969C1EA568DBDEF583BB0403090B500C362580B39930AF3F692D58027A197671A3724897F1F2FC1CBBACE70FD0723D5548DE7E5F11ADD1BD7288FBA7D2B54E3C754E8CA8557B5D1E49D3B3D95187ED75EDE3254516E80038CD415A283563473B98B1A436FECDA3322F6019E176E0CF1D7B16CF23B7147DBD8A147543712EC5453F0C513B883D559DF667681ACA9878F8DD7FA3737836D942D14877D5784AFC8A12FE904CC88B3D6B4C37A28FF391078BD22DBB88E76E68675017A6DFDA37F143F53D28F8BA1FD1A1250247F231F544523D4AFE9FCC1FD2363D1B47AFFE64BFA62810A273F43B5C7FDA86131A0152B4FEC3FD54A2E24A1A008CF05DBFF530969B113C67484DADEF2A7328D9D5E389209AF767629D623DA8433175DFA650D6CD4C6903FCA62D4F8AF2FA7A2C0FAE7DBB3479DD35B3956EA1F2A508C955238D53E96D4EC2A73632F4FE019D971D0F887F2B830970BF00F858E6534731025F10ABD02B4AAF8BC1E441981F23D2B833535D0379D3F2C12497401425C8A3DA8E58C14BE9BE2664B6596C700"; // Replace with your preset cookie

    $audioUrl = fetchAudioLocation($assetId, $placeId, $robloxCookie);
    $assetName = fetchAssetName($assetId); // Fetch asset name

    if ($audioUrl) {
        echo '<h3>' . $assetName . '</h3>'; // Echo asset name
        echo '<video controls autoplay name="media">';
        echo '<source src="' . $audioUrl . '" type="audio/ogg">';
        echo '</video>';
    } else {
        echo "Audio location not found.";
    }
}

function fetchAudioLocation($assetId, $placeId, $robloxCookie) {
    while (true) {
        try {
            $bodyArray = [
                [
                    "assetId" => $assetId,
                    "assetType" => "Audio",
                    "requestId" => "0"
                ]
            ];

            $headers = [
                "User-Agent: Roblox/WinInet",
                "Content-Type: application/json",
                "Cookie: .ROBLOSECURITY=" . $robloxCookie,
                "Roblox-Place-Id: " . $placeId,
                "Accept: */*",
                "Roblox-Browser-Asset-Request: false"
            ];

            $ch = curl_init();
            curl_setopt($ch, CURLOPT_URL, "https://assetdelivery.roblox.com/v2/assets/batch");
            curl_setopt($ch, CURLOPT_POST, true);
            curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
            curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($bodyArray));
            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
            $response = curl_exec($ch);
            curl_close($ch);

            if ($response) {
                $locations = json_decode($response, true);
                if ($locations && count($locations) > 0) {
                    $obj = $locations[0];
                    if (isset($obj["locations"]) && isset($obj["locations"][0]["location"])) {
                        $audioUrl = $obj["locations"][0]["location"];
                        return $audioUrl;
                    }
                }
            }
        } catch (Exception $error) {
            error_log($error);
        }

        // Simulate waiting before retrying
        usleep(500000); // 500 milliseconds
    }
}

function fetchAssetName($assetId) {
    while (true) {
        $response = file_get_contents("https://economy.roproxy.com/v2/assets/{$assetId}/details");
        if ($response) {
            $assetInfo = json_decode($response, true);
            $assetName = $assetInfo["Name"];
            if ($assetName) {
                return $assetName;
            }
        }
        usleep(500000); // 500 milliseconds
    }
}
?>
